### PR TITLE
#13030: Eliminate namespace pollution

### LINKED ISSUE #13030: Avoid 'using' in header files; Namespace pollution
https://github.com/tenstorrent/tt-metal/blob/0adc6246d6acd2e62f23d48b25932d094c2a42fe/tt_metal/common/base.hpp#L24-L32

### PR DESCRIPTION
**Pull request by blozano-tt**
### Ticket
closes #13030
#596 

### Problem description
`tt_metal/common/base.hpp` is a leaf level include to almost every translation unit in the whole project.
It is being used as a `catch_all` header, that brings several things into all TU.
It causes every TU to require `ARCH_NAME` related include paths, because of what it brings in.
It has legacy unused code.
It follows poor practice, of namespace pollution, by polluting every TU with `using` keyword.

### What's changed
I removed almost everything from `tt_metal/common/base.hpp`.
I fixed the namespace pollution in all impacted TU.
I removed the dead code.

### Checklist
https://github.com/tenstorrent/tt-metal/actions/runs/11526634995
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]
Leaving x10 APPROVE

**Review by sminakov-tt — APPROVED** [review]

**Review by dmakoviichuk-tt — APPROVED** [review]

**Review by SeanNijjar — APPROVED** [review]

File: tt_metal/common/base.hpp [hunk 1]
```diff
@@ -7,56 +7,22 @@
...
-using std::ostream;
-using std::uint8_t;
-using std::uint32_t;
-using std::uint64_t;
-using std::vector;
-using std::string;
-using std::size_t;
-using std::map;
+// DO NOT ADD MORE CODE TO THIS FILE
+// THIS FILE POLLUTES ALL TRANSLATION UNITS - tt_metal, ttnn, programming examples, tests, customer code
```
Comment (afuller-TT) [inline tt_metal/common/base.hpp:13]: ❤️

**Review by afuller-TT — APPROVED** [review]
Yes yes please yes!!

Incoming breaking change for customers, but better now than later!

**Comment by pavlejosipovic** [comment]
@blozano-tt @afuller-TT 
Hi guys I didn't get a chance to comment on this one, but it seems like this change is making build times for local development even worse as you kicked out few header from pch.
Did any of you measure the impact of this change?

**Pull request merged**